### DISCUSSION TITLE
8Bit MCU8080 Parallel display driver

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#11036] [ESP32] [answered]
Hello, I would like to use MicroPython on the WT32-SC01 Plus (combo board with an ESP32-S3 and a 3.5 touch LCD panel) which uses 8 bit parallel for the display (driver ST7796UI).
Any ideas which libraries that could be used to use it with MicroPython?
Thanks

**Comment by russhughes** [comment]
Here is a link to a display driver written in MicroPython [wt32sc01py](https://github.com/russhughes/wt32sc01py). I'm working on a faster driver written in 'C,' but I'm still having issues with the DMA.

**Comment by russhughes** [comment, marked as answer]
Here is a link to a faster driver written in C with a precompiled firmware image for the WT32-SC01 Plus [wt32_esp_lcd](https://github.com/russhughes/wt32_esp_lcd). This driver uses the [esp-idf esp_lcd component](https://docs.espressif.com/projects/esp-idf/en/v4.4.4/esp32/api-reference/peripherals/lcd.html) and requires a framebuffer the same size as the screen.

**Reply by saraverbeecke** [reply]
Great tip! Thanks. I will try both of your suggestions. Do you know if this could work with LVGL as well (if I build both the C libraries into the firmware with MicroPython)?

**Comment by russhughes** [comment]
It would not work with LVGL in its current form. The first issue is to get the LVGL MicroPython to compile for the ESP32S3 using a new enough esp_idf version to support the esp_lcd component. Once solved, you would only need the code from the driver that initializes and updates the display. 

See https://github.com/lvgl/lv_binding_micropython/issues/227 for more information.

**Reply by saraverbeecke** [reply]
OK thanks for showing the way Russ!

**Comment by kdschlosser** [comment]
I have written a way that uses the gen_mpy.py script in the lv_micropython binding that exposes all of the esp_lcd component functions and structures to MicroPython..

I am working on a way to make it easy to use when compiling MicroPython. Currently there is no easy way to add components to the component list from the command line. That would be the ideal way to do it. Currently it is able to espose the esp_lcd, esp_hid and usb component and also twai. not sure what else could be added to that list. Maybe the esp_now parts of the esp_wifi component.
